SWLOR.Game.Server/Event/Module/OnModuleLeave.cs

```diff
@@ -34,6 +34,11 @@ public bool Run(params object[] args)
         {
             NWPlayer pc = (_.GetExitingObject());
 
+            if (pc.IsDM)
+            {
+                App.GetAppState().ConnectedDMs.Remove(pc);
```

Comment:
In general I *prefer* to inject the AppState via constructor (just add AppState to constructor, add a member variable and hook it up there). This is essentially using the service locator which is widely considered to be an anti-pattern. The only reason this exists is because I had to break the rules a bit when I removed all of the dependencies out of NWObject.